kornia/utils/draw.py

```diff
@@ -205,3 +205,72 @@ def draw_rectangle(
                 ]
 
     return image
+
+
+def _get_convex_edges(polygon: torch.Tensor, h: int, w: int):
+    r"""Gets the left and right edges of a polygon for each y-coordinate y \in [0, h)
+    Args:
+        polygon: represents polygons to draw in BxNx2
+            N is the number of points
+            2 is (x, y).
+        h: bottom most coordinate (top coordinate is assumed to be 0)
+        w: right most coordinate (left coordinate is assumed to be 0)
+    Returns:
+        The left and right edges of the polygon of shape Bx2.
+    """
+
+    if not torch.allclose(polygon[..., -1, :], polygon[..., 0, :]):
+        polygon = torch.cat((polygon, polygon[..., :1, :]), dim=-2)
+    x_start = polygon[..., :-1, 0]
+    x_end = polygon[..., 1:, 0]
+    y_start = polygon[..., :-1, 1]
+    y_end = polygon[..., 1:, 1]
+    ys = torch.arange(h, device=polygon.device)
+    dx = torch.clamp((x_end - x_start) / (y_end - y_start), -w, w)
```

Comment:
```suggestion
    dx = ((x_end - x_start) / (y_end - y_start)).clamp(min=-w, max=w)
```
More pythonic, to easily later dynamically rename to .clip ;)

Reply:
Fixed